core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -162,11 +178,13 @@ void delete(F file) {
   void delete(CharSequence path) {
     Preconditions.checkNotNull(path, "Cannot delete file path: null");
     invalidateFilteredCache();
+    this.allDeletesReferenceManifests = false;
     deletePaths.add(path);
   }
 
   boolean containsDeletes() {
-    return !deletePaths.isEmpty()
+    return !manifestsReferencedForDeletes.isEmpty()
```

Comment:
Is this change actually required? Won't checking `deleteFiles` below be sufficient?
We don't check `deleteFilePartitions`, for instance.

Reply:
It's not strictly required since we currently always add to `deleteFiles` when deleting by the content file, I can remove it if you think it's not as readable to have it. 

I think I'd prefer to keep it though because in case some future logic shifts around for whatever reason, containsDeletes() will just return the right value instead of us having to make sure we update this method.

But I'm good either way, not super opinonated on it!

Reply:
This also stood out to me. While it is more defensive, it warrants at least a comment. My concern is that a future maintainer gets confused because any referenced manifest implies that there are deletes. That's not in itself true. It's true in the current implementation.

I would prefer to remove this.

Reply:
I've removed this